presto-jdbc/src/main/java/io/prestosql/jdbc/PrestoConnection.java

```diff
@@ -92,6 +97,7 @@
     private final Map<String, String> sessionProperties = new ConcurrentHashMap<>();
     private final Map<String, String> preparedStatements = new ConcurrentHashMap<>();
     private final Map<String, SelectedRole> roles = new ConcurrentHashMap<>();
+    private final Map<String, String> extraCredentials = new ConcurrentHashMap<>();
```

Comment:
Move this above, after `user`, and remove the initializer. The value should be a constant, since it's set in the constructor and we don't modify it afterwards.